server/sonar-qa-util/src/main/java/org/sonarqube/qa/util/pageobjects/Navigation.java

```diff
@@ -241,31 +235,44 @@ public RulesPage openRules() {
   }
 
   public SelenideElement clickOnQualityProfiles() {
-    return $(By.linkText("Quality Profiles"));
+    return Selenide.$(By.linkText("Quality Profiles"));
   }
 
   public SelenideElement getRightBar() {
-    return $("#global-navigation .navbar-right");
+    return Selenide.$("#global-navigation .navbar-right");
   }
 
   public SelenideElement getFooter() {
-    return $("#footer");
+    return Selenide.$("#footer");
   }
 
   public SelenideElement getErrorMessage() {
-    return $("#error");
+    return Selenide.$("#error");
+  }
+
+  private static SelenideElement logInLink() {
+    return Selenide.$(By.linkText("Log in"));
   }
 
-  private SelenideElement logInLink() {
-    return $(By.linkText("Log in"));
+  private static SelenideElement loggedInDropdown() {
+    return Selenide.$(".js-user-authenticated");
   }
 
-  private SelenideElement loggedInDropdown() {
-    return $(".js-user-authenticated");
+  /**
+   * Safe encoding for  URL parameters
+   * @param parameter the parameter to escape value
+   * @return the escaped value of parameter
+   */
+  private static String escape(String parameter) {
+    try {
+      return URLEncoder.encode(parameter, "UTF-8");
+    } catch (UnsupportedEncodingException e) {
+      throw new IllegalArgumentException(format("Unable to escape [%s]", parameter));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00112)